training-service/src/app/content/category/component/category-name/category-name.component.ts

```diff
@@ -0,0 +1,18 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+
+@Component({
+  selector: 'app-category-name',
+  templateUrl: './category-name.component.html',
+  styleUrls: ['./category-name.component.scss']
+})
+export class CategoryNameComponent implements OnInit {
+
+  name: string | undefined;
+
+  constructor(private route: ActivatedRoute) { }
+
+  ngOnInit() {
+    this.route.params.subscribe(p => this.name = p['name']);
```

Comment:
Unsubscribe is missing for this subscription, this is a potential memory leak.

Reply:
done